src/metabase/formatter.clj

```diff
@@ -127,22 +129,25 @@
                                            decimals-in-value))))
               fmt-str (cond-> base
                         (not (zero? decimal-digits)) (str "." (apply str (repeat decimal-digits "0")))
-                        scientific? (str "E0")
-                        percent?    (str "%"))
+                        scientific? (str "E0"))
               fmtr (doto (DecimalFormat. fmt-str symbols) (.setRoundingMode RoundingMode/HALF_UP))]
           (map->NumericWrapper
            {:num-value value
-            :num-str   (str (when prefix prefix)
-                            (when (and currency? (or (nil? currency-style)
-                                                     (= currency-style "symbol")))
-                              (get-in currency/currency [(keyword (or currency "USD")) :symbol]))
-                            (when (and currency? (= currency-style "code"))
-                              (str (get-in currency/currency [(keyword (or currency "USD")) :code]) \space))
-                            (cond-> (.format fmtr scaled-value)
-                              (not decimals) (strip-trailing-zeroes decimal))
-                            (when (and currency? (= currency-style "name"))
-                              (str \space (get-in currency/currency [(keyword (or currency "USD")) :name_plural])))
-                            (when suffix suffix))}))
+            :num-str   (let [inline-currency? (and currency?
+                                                   (false? (::mb.viz/currency-in-header column-settings)))]
+                         (str (when prefix prefix)
+                              (when (and inline-currency? (or (nil? currency-style)
+                                                       (= currency-style "symbol")))
+                                (get-in currency/currency [(keyword (or currency "USD")) :symbol]))
+                              (when (and inline-currency? (= currency-style "code"))
+                                (str (get-in currency/currency [(keyword (or currency "USD")) :code]) \space))
+                              (cond-> (.format fmtr scaled-value)
+                                (and (not currency?) (not decimals))
+                                (strip-trailing-zeroes decimal)
+                                percent?    (str "%"))
```

Comment:
I can't quite follow why this `(str "5")` was moved from the `fmt-str` in the let binding? 
Whatever the logic is for moving it, should it also be applied to the `scientific?` pred that is still in `fmt-str`?

Reply:
This was done so that trailing zeros could be trimmed from the string. This is to be consistent with the FE.

Reply:
Ah, of course! Thanks